### PR TITLE
Enable make check (unit tests only) and check for warnings in travis

### PR DESCRIPTION
**Pull request by dlespiau**
Another step towards better CI:

- Get rid of warnings and make sure we don't introduce any later on
- Run make check

What's still to do is documented in https://github.com/01org/cc-oci-runtime/issues/69

### CODE REVIEW

File: src/commands/create.c [hunk 1]
```diff
@@ -22,6 +22,10 @@
 
 extern struct start_data start_data;
 
+/* ignore -pedantic to cast handle_option_console, a function pointer, to a
+ * void* */
+#pragma GCC diagnostic push
```
Comment (grahamwhaley) [inline src/commands/create.c]: I'd like to hear @jodh-intel opinion on the pragmas. Great that you've nullified the warnings, but in-code GCC specific pragmas are not generally to my liking :-)  If James is OK, then I think we'll probably be fine.

File: tests/hypervisor_test.c [hunk 2]
```diff
@@ -524,12 +524,12 @@ START_TEST(test_cc_oci_vm_args_get) {
 	g_strfreev (args);
 
 	/* recreate the args file */
-	ret = g_file_set_contents (args_file, "hello\# comment\n", -1, NULL);
+	ret = g_file_set_contents (args_file, "hello\\# comment\n", -1, NULL);
```
Comment (grahamwhaley) [inline tests/hypervisor_test.c]: Good catch - but then this hints to me that the test was not really working or maybe testing what it should... @devimc can you have a think about that?

**Comment by dlespiau** [comment]
Don't really want to open the pandora box, but I'd drop pedantic for -Wall and either all or some of the -Wextra warnings. I have yet to find an actually bug -pedantic helped catching and considering we're only targeting linux and x86_64, seems overkill to me.

**Comment by grahamwhaley** [comment]
Nice series, thanks for doing that. Overall it looks good to me. Just a couple of minor comments.

File: src/commands/create.c (hunk 1 above)
Comment (jodh-intel) [inline src/commands/create.c]: @grahamwhaley - I agree, not generally a fan of pragmas, but aside from changing `-std=`, I don't know how we can suppress these warnings.

Other thoughts on pragmas / options to suppress:

- will static analysis tools understand them?
- will other compilers understand them?

**Comment by jodh-intel** [comment]
To be clear, I'm +1 too. We might want to revisit the use of `-pedantic` at some point, but for now the ability to have Travis running the tests is more important IMHO. Added to which the pragma is commented and they are easy to `grep` :)

**Comment by grahamwhaley** [comment]
OK, I was about to merge this and let @devimc pick up on the \\# test item above separately - but we now have merge conflicts. @dlespiau , if you could fix that then we'll merge.  Thanks.

**Comment by dlespiau** [comment]
A few changes:

- Rebased on top of master
- Added a couple more commits to compile warning free with clang
- Changed the pragma to ignore "-Wpedantic" instead of "-pedantic" as the latter wasn't working with clang (I usually include a change log in the commit message)

File: src/commands/create.c (hunk 1 above)
Comment (dlespiau) [inline src/commands/create.c]: I guess the answers to those questions will have to come with time. Worse than can happen is being ignored though, maybe with a warning.

The good news is that it's now also working with clang.

**Comment by dlespiau** [comment]
Ok, passing again. This time it was and interaction with the parallel build (-j5) that just got added, patch has the details.

**Pull request merged**